agent/token_store_test.go

```diff
@@ -0,0 +1,97 @@
+package agent
+
+import (
+	"testing"
+)
+
+func TestTokenStore_AgentToken(t *testing.T) {
+	t.Parallel()
+	tokens := new(TokenStore)
+
+	if got, want := tokens.GetTokenForAgent(), ""; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	// Precedence is agent token, user token, "".
+
+	tokens.UpdateUserToken("user")
+	if got, want := tokens.GetTokenForAgent(), "user"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateAgentToken("agent")
+	if got, want := tokens.GetTokenForAgent(), "agent"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateUserToken("")
+	if got, want := tokens.GetTokenForAgent(), "agent"; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+
+	tokens.UpdateAgentToken("")
+	if got, want := tokens.GetTokenForAgent(), ""; got != want {
+		t.Fatalf("got %q want %q", got, want)
+	}
+}
+
+func TestTokenStore_UserToken(t *testing.T) {
```

Comment:
If you rewrite the previous test as a table driven test then you can roll this test in there as well.